core/src/test/java/io/jenkins/pluginhealth/scoring/probes/HasUnreleasedProductionChangesProbeTest.java

```diff
@@ -0,0 +1,271 @@
+                assertThat(commit.getFullMessage().equals("Imports pom.xml file")).isTrue();
+                assertThat(timestampInstant.isBefore(dateInstant)).isEqualTo(true);
+            }
+        }
+    }
+
+    @Test
+    void commitOnReadmeFileAfterReleaseDateShouldReturnSuccess() throws IOException, GitAPIException {
+        final Path repository = Files.createTempDirectory("test-foo-bar");
+
+        final Plugin plugin = mock(Plugin.class);
+        final ProbeContext ctx = mock(ProbeContext.class);
+
+        ZonedDateTime releaseTimestamp = ZonedDateTime.now();
+        when(plugin.getReleaseTimestamp()).thenReturn(releaseTimestamp);
+
+        when(plugin.getDetails()).thenReturn(Map.of(
+            SCMLinkValidationProbe.KEY, ProbeResult.success(SCMLinkValidationProbe.KEY, "")
+        ));
+        when(ctx.getScmRepository()).thenReturn(repository);
+
+        try (Git git = Git.init().setDirectory(repository.toFile()).call()) {
+
+            final Path pom = Files.createFile(repository.resolve("README.md"));
+
+            // creating commit
+
+            PersonIdent defaultCommitter = new PersonIdent(git.getRepository());
+            PersonIdent committer = new PersonIdent(defaultCommitter, Date.from(plugin.getReleaseTimestamp().plusDays(1).toInstant()));
+
+            git.add().addFilepattern("README.md").call();
+            git.commit().setMessage("Updated README.md file").setSign(false).setCommitter(committer).call();
+
+            final HasUnreleasedProductionChangesProbe probe = getSpy();
+
+            for (RevCommit commit : git.log().call()) {
+                long timestamp = commit.getCommitTime();
+                String dateString = plugin.getReleaseTimestamp().toString();
+
+                Instant timestampInstant = Instant.ofEpochSecond(timestamp);
+                DateTimeFormatter formatter = DateTimeFormatter.ISO_ZONED_DATE_TIME;
+                ZonedDateTime zonedDateTime = ZonedDateTime.parse(dateString, formatter);
+                Instant dateInstant = zonedDateTime.toInstant();
+
+                final ProbeResult result = probe.apply(plugin, ctx);
+                assertThat(commit.getFullMessage().equals("Updated README.md file")).isTrue();
+                assertThat(dateInstant.isBefore(timestampInstant)).isEqualTo(true);
+            }
+        }
+    }
+
+    @Test
+    void checkThatCommitOnSrcPathBeforeReleaseDateReturnsSuccess() throws IOException, GitAPIException {
+        final Path repository = Files.createTempDirectory("test-foo-bar");
+
+        final Plugin plugin = mock(Plugin.class);
+        final ProbeContext ctx = mock(ProbeContext.class);
+
+        final HasUnreleasedProductionChangesProbe hasUnreleasedProductionChangesProbe = mock(HasUnreleasedProductionChangesProbe.class);
+        when(hasUnreleasedProductionChangesProbe.apply(plugin, ctx)).thenReturn(ProbeResult.success("unreleased-production-changes", ""));
```

Comment:
As said in gitter, the probe shouldn't be a mock. Other tests are using a `spy` because it was required to have "common" tests done by the `AbstractProbeTest`. 
However, `spy` are using the real class code. 

More important is that you are saying "when I call the probe `apply` method, with this plugin and context, I should get a success". This is a problem because is this exactly what you want to validate and here you are not validating the probe code, you are bypassing it.